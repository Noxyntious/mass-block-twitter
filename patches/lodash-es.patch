diff --git a/_root.js b/_root.js
index 281f81280ec9ea137b644ba60b9f5c6ccede4f88..c742f2c2f313aa0a07c8df371e818a3702634b9f 100644
--- a/_root.js
+++ b/_root.js
@@ -4,6 +4,6 @@ import freeGlobal from './_freeGlobal.js';
 var freeSelf = typeof self == 'object' && self && self.Object === Object && self;
 
 /** Used as a reference to the global object. */
-var root = freeGlobal || freeSelf || Function('return this')();
+var root = freeGlobal || freeSelf || globalThis;
 
 export default root;
