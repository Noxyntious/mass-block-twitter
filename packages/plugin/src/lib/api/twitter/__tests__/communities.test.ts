import { describe, expect, it } from 'vitest'
import {
  CommunityInfo,
  parseCommunityInfo,
  parseCommunityMembers,
} from '../communities'
import CommunityQuery from './assets/CommunityQuery.json'
import MembersSliceTimelineQuery from './assets/MembersSliceTimeline_Query.json'
import { omit } from 'lodash-es'

describe('parseCommunityMembers', () => {
  it('parseCommunityMembers', () => {
    const users = parseCommunityMembers(MembersSliceTimelineQuery)
    expect(users.data).length(20)
    expect(users.data.every((it) => it.id && it.community_role)).true
    expect(users.cursor).not.undefined
    expect(users.cursor).toMatchSnapshot()
    expect(
      users.data.map((it) => omit(it, 'created_at', 'updated_at')),
    ).toMatchSnapshot()
  })
})

describe('parseCommunityInfo', () => {
  it('parseCommunityInfo', () => {
    const info = parseCommunityInfo(CommunityQuery)
    expect(info).toEqual({
      id: '1900366536683987325',
      name: 'Misa',
      description: 'Misa',
      member_count: 6700,
      is_nsfw: false,
    } satisfies CommunityInfo)
  })
})
