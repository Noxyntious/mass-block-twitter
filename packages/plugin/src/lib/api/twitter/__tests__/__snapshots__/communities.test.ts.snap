// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseCommunityMembers > parseCommunityMembers 1`] = `"CycRAAAAAAwABAwKmwzBVQo+vQ1AIqAVF9ABCm07AAABlZJtlFcADJorCj69f/////////8KbTt//////////wAIP8AAAAACAAAA"`;

exports[`parseCommunityMembers > parseCommunityMembers 2`] = `
[
  {
    "blocking": true,
    "community_role": "Admin",
    "following": false,
    "id": "1759646691316105216",
    "name": "Tonio",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1825654889688481792/AGw5QSoa_normal.jpg",
    "screen_name": "TonioAbrazo",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "1815467469353439232",
    "name": "Joaconel 2.0",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1902518416553934848/W96aCtib_normal.jpg",
    "screen_name": "Joaconel_2",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "1511730907786498051",
    "name": "Taliban üá¶üá∑",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1625358685026369536/Wj1JLHSg_normal.jpg",
    "screen_name": "TalibanEzquiso",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "1568007265252253720",
    "name": "Javier Miller",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1886505156549718016/CX6bkOU4_normal.jpg",
    "screen_name": "GordoChoto_",
  },
  {
    "blocking": true,
    "community_role": "Moderator",
    "following": false,
    "id": "1373123472",
    "name": "Mort√≠fago del Kraken",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1802552118953353216/BHbrhQZq_normal.jpg",
    "screen_name": "BrianLPBlue",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "1541654566160285697",
    "name": "Armenthius",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1734349161003204608/x6jseVmG_normal.jpg",
    "screen_name": "Armenthius",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "914644011406839813",
    "name": "Troll de Ceniza",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1840068929474936832/31-HCYNP_normal.jpg",
    "screen_name": "trollantikarg",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "1761857318465994752",
    "name": "Se√±or Fr√≠o üå®Ô∏è‚õÑüèÇ",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1784313581728587776/OZGTfag3_normal.jpg",
    "screen_name": "doc_frio",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "762779995588726784",
    "name": "Facu Gonzalez",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1746687055889944576/vvYGKQiy_normal.jpg",
    "screen_name": "RevanHD123",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "1844212227814940683",
    "name": "Manuel Garc√≠a Gojon ü™ô",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1844212424406122498/y0kZCtrh_normal.jpg",
    "screen_name": "ManuGarGojMX",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "4870970092",
    "name": "Valen üö¨üêçüá¶üá∑",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1873962574699466752/p7Xqx4Or_normal.jpg",
    "screen_name": "valen_A_C_",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "3038292197",
    "name": "Longinuss__",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1840740290291699712/O7hm91Bd_normal.jpg",
    "screen_name": "Longinuss_",
  },
  {
    "blocking": false,
    "community_role": "Moderator",
    "following": false,
    "id": "198905691",
    "name": "Pablo",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1697688444124680192/iPhLGOks_normal.jpg",
    "screen_name": "MenemAbrazo",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "3291372697",
    "name": "Sofi Lo Piccolo",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1819746161562980352/99GcaHaK_normal.jpg",
    "screen_name": "sofilopiccolo",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "2523570561",
    "name": "Lu",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1832444514083934208/OAr4VpIy_normal.jpg",
    "screen_name": "lu_se_7",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "1325132244634120196",
    "name": "SobrinoGü¶Å‚öïÔ∏è",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1693410711597150208/GWqoWj88_normal.jpg",
    "screen_name": "SobrinoG_333",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "1392331667445014534",
    "name": "Ero Cook",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1392332280325156871/wyYIo3Kc_normal.jpg",
    "screen_name": "EroCook4",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "1186765210373312533",
    "name": "Pappo",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1879655282906738688/g7HrfJfy_normal.jpg",
    "screen_name": "Bruno63697253",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "1839441831790276608",
    "name": "m*adelcielo",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1839442704192995329/gMaZ896E_normal.jpg",
    "screen_name": "Tuiterodelcielo",
  },
  {
    "blocking": true,
    "community_role": "Member",
    "following": false,
    "id": "954801191946539008",
    "name": "Aoife (Irish Version)",
    "profile_image_url": "https://pbs.twimg.com/profile_images/1529298587091230720/yVPgQPHq_normal.jpg",
    "screen_name": "iwannabeapurse",
  },
]
`;
